core/unit_test/TestComplex.hpp

```diff
@@ -355,4 +355,24 @@ TEST(TEST_CATEGORY, complex_special_funtions) {
 
 TEST(TEST_CATEGORY, complex_io) { testComplexIO(); }
 
+TEST(TEST_CATEGORY, complex_trivially_copyable) {
+  using RealType = double;
+
+  // Kokkos::complex<RealType> is trivially copyable when RealType is trivially
+  // copyable
+
+#if defined(KOKKOS_COMPILER_GNU) && KOKKOS_COMPILER_GNU < 500
```

Comment:
This will hit all compilers that use the GCC header files. SO we need to to check for the gcc header file intrinsic.

Reply:
I think this just needs to be `#if defined(__GNUC__) && __GNUC__ < 5`

Reply:
OK and larger than 0 for clang?